### PR TITLE
Support `in_channels` with `tuple` in `GENConv` for bipartite message passing

### PR DESCRIPTION
**Pull request by lightaime**
This PR supports passing `in_channels` with `tuple` to `GENConv` for bipartite message passing by projecting channels to the same dimension if channels mismatch.

### CODE REVIEW

File: torch_geometric/nn/conv/gen_conv.py [hunk 1]
```diff
@@ -62,8 +62,8 @@ class GENConv(MessagePassing):
         ogbn_proteins_deepgcn.py>`_.
 
     Args:
-        in_channels (int): Size of each input sample, or :obj:`-1` to derive
-            the size from the first input(s) to the forward method.
+        in_channels (int or tuple): Size of each input sample, or :obj:`-1` to
+            derive the size from the first input(s) to the forward method.
```
Comment (rusty1s) [inline torch_geometric/nn/conv/gen_conv.py:66]:
```suggestion
        in_channels (int or tuple): Size of each input sample, or :obj:`-1` to
            derive the size from the first input(s) to the forward method.
            A tuple corresponds to the sizes of source and target
            dimensionalities.
```

File: torch_geometric/nn/conv/gen_conv.py [hunk 2]
```diff
@@ -101,41 +101,65 @@ class GENConv(MessagePassing):
...
+
+        channels = [in_channels[0]]
         for i in range(num_layers - 1):
-            channels.append(in_channels * 2)
+            channels.append(in_channels[0] * 2)
         channels.append(out_channels)
         self.mlp = MLP(channels, norm=norm)
 
-        self.msg_norm = MessageNorm(learn_msg_scale) if msg_norm else None
+        if in_channels[0] != in_channels[1]:
```
Comment (rusty1s) [inline torch_geometric/nn/conv/gen_conv.py:150]: and `in_channels=[0] != -1?

**Review by rusty1s — APPROVED** [review]
Can we add some corresponding tests?

File: torch_geometric/nn/conv/gen_conv.py (hunk 2 above)
Comment (lightaime) [reply torch_geometric/nn/conv/gen_conv.py]: Good point!

**Comment by lightaime** [comment]
> Can we add some corresponding tests?

Sure. Will update it later.

**Pull request merged**